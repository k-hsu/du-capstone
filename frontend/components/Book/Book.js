import React from "react";
import styled from "@emotion/styled";
import { useRouter } from "next/router";
import Text from "../Text/Text";
import { color, fontWeight, spacing } from "../../theme";

const StyledBook = styled.div`
  display: flex;
  flex-direction: column;
  width: 250px;
  padding: ${spacing["2.5"]} ${spacing["0.75"]};
  border: 1px solid ${color.white};
  cursor: pointer;
`;

const Book = ({ id, author: { firstName, lastName }, title }) => {
  const router = useRouter();

  const onBookClick = (event) => {
    if (!event.key || event.key === "Enter") {
      router.push(`/books/${id}`);
    }
  };
  return (
    <StyledBook tabIndex={0} onClick={onBookClick} onKeyDown={onBookClick}>
      <Text as="h2" lineHeight={0}>
        {title}
      </Text>
      <Text
        as="h2"
        lineHeight={0}
        fontWeight={fontWeight.normal}
        fontStyle="italic"
      >{`${firstName} ${lastName}`}</Text>
    </StyledBook>
  );
};

export default Book;
