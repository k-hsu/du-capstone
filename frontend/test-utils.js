import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
