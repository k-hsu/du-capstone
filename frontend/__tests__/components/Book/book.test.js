import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Book from "../../../components/Book/Book";

jest.mock("next/router", () => ({
  useRouter: () => ({ push: () => {} }),
}));
describe("Book", () => {
  it("should render book component", () => {
    render(
      <Book
        title="Harry Potter and the Chamber of Secrets"
        author={{ firstName: "J.K.", lastName: "Rowling" }}
      />
    );
    expect(
      screen.getByText("Harry Potter and the Chamber of Secrets")
    ).toBeInTheDocument();
    expect(screen.getByText("J.K. Rowling")).toBeInTheDocument();
  });
  it("should render book component styles", () => {
    const component = renderer.create(
      <Book
        title="Harry Potter and the Chamber of Secrets"
        author={{ firstName: "J.K.", lastName: "Rowling" }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
