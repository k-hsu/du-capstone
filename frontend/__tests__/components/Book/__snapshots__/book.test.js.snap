// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render book component styles 1`] = `
<div
  className="css-xi7dgk"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <h2
    className="css-c0iozm"
    fontFamily="IBM Plex Sans"
  >
    Harry Potter and the Chamber of Secrets
  </h2>
  <h2
    className="css-t3u8wj"
    fontFamily="IBM Plex Sans"
    fontStyle="italic"
    fontWeight="400"
  >
    J.K. Rowling
  </h2>
</div>
`;
